Update WebGPU onsubmittedworkdone (#27231)
mdn/content: files/en-us/web/api/gpuqueue/onsubmittedworkdone/index.md

This includes the completion of any {{domxref("GPUBuffer.mapAsync", "mapAsync()")}} calls made on `GPUBuffer`s used in commands submitted to the queue, before `onSubmittedWorkDone()` is called.

> Note: In most cases, you do NOT need to call `onSubmittedWorkDown()`. You do **NOT** need to call it for mapping a buffer. `mapAsync` guarantees work submitted
> to the queue before calling `mapAsync` happens before the `mapAsync` returns (see [WebGPU spec: section 5.2](https://www.w3.org/TR/webgpu/#buffer-mapping))

Reviewer: ```suggestion
Note: In most cases, you do _not_ need to call `onSubmittedWorkDown()`. You do **_not_** need to call it for mapping a buffer. `mapAsync` guarantees work submitted
to the queue before calling `mapAsync` happens before the `mapAsync` returns (see [WebGPU spec: section 5.2](https://www.w3.org/TR/webgpu/#buffer-mapping))
```

> The 2 use-cases for `onSubmittedWorkDone`

Reviewer: ```suggestion
The two use-cases for `onSubmittedWorkDone`
```

> The two use-cases for `onSubmittedWorkDone`

Reviewer: ```suggestion
The two use cases for `onSubmittedWorkDone`
```

1. Waiting for multiple buffer mapping (slow)

   ```js
   // good
   await Promise.all([
     buffer1.mapAsync(),
     buffer2.mapAsync(),
     buffer3.mapAsync(),
   ]);
   data1 = buffer1.getMappedRange();
   data2 = buffer2.getMappedRange();
   data3 = buffer3.getMappedRange();
   ```

   ```js
   // works but slow
   buffer1.mapAsync();
   buffer2.mapAsync();
   buffer3.mapAsync();
   await device.queue.onSubmittedWorkDone();
   data1 = buffer1.getMappedRange();
   data2 = buffer2.getMappedRange();
   data3 = buffer3.getMappedRange();
   ```

>    The reason the 2nd method is slow is the implementation may be able to map the buffers before all the submitted work is done.

Reviewer: ```suggestion
   The reason the second method is slow is, the implementation may be able to map the buffers before all the submitted work is done.
```

>    For example, if all the buffers are finished being used but more work, un-related to the buffers, is already submitted then

Reviewer: ```suggestion
   For example, if all the buffers are finished being used, but more work (unrelated to the buffers) is already submitted, then
```
